skel/lib/systemd/system-generators/dcache-generator

```diff
@@ -4,6 +4,10 @@ set -e
 
 @DCACHE_LOAD_CONFIG@
 
+
+wantdir="$1/dcache.service.wants"
+mkdir "$wantdir"
```

Comment:
You may want to add a `-p` to this.

Reply:
Debian had that as well, IIRC, but I'm sceptical.... the dir should already be there and if not, something else is wrong in the system,... and we shouldn't try to hide the other problem in that case.

Reply:
You mean should NOT be there? Because if it already exists from a previous run, then your command would fail. That's why I suggested adding `-p`. Are you certain that systemd deletes the dir before rerunning the generator after a daemon reload?

Reply:
I meant, the parent dirs should already be there (created by systemd)... AFAIU, the dir itself (i.e. the wantdir) should NOT be there cause my understanding was, that whenever the generators run, the whole /run/.../ stuff is flushed before.

If it was not (or if there was some other situation, in which the generators run, but the whole stuff isn't flushed away before), then I think it would be actually good for us to notice the failure. Why?
Because we "depend" on the flushing to get rid of e.g. de-configured domains in such cases.

This (having the clean up of the generated stuff) is also the reason why I want dCache's cached config to move away from /var/lib/dcache/config ...
Cause *if* we shall make use of generated per-domain EnvironmentFile= ... than these also need to be cleaned up at least from eventually (not necessarily on any system reload though)... having that stuff in /run/ which is typically tmpfs, makes me feel better ;-)

Reply:
Yeah, of course the parent directory is there. As long as you are certain the wantdir does not exist, then fine with me. What I was not certain about was whether the flushing would just delete the files/links or also the directories.

And if you want to move the config cache, be my guess. That stuff was written many years ago.

Reply:
Well at least it works for me on 3.2 (on a test dcache)... even when reloading systemd multiple times.
I have no strong opinion about the -p ... if someone wants it ... I'd add it... if not the benefit is that we notice if it doesn't get flushed as we assume anyway :-)

As for the moving of the config cache... I tried to... but this seems to be coded in many places in dCache... so it's not done by changing path.properties... but I guess one needs to hack it into the build system.. and I didn't dare to touch this O:-)